src/updateFileFromServer.ts

```diff
@@ -73,12 +122,53 @@ export async function toggleServerTaskStatus(e: Editor, settings: TodoistSetting
 	}
 }
 
-
-async function getServerData(todoistQuery: string, authToken: string): Promise<string> {
+async function getServerData(todoistQuery: string, authToken: string, showSubtasks: boolean): Promise<string> {
 	const api = new TodoistApi(authToken)
+
+	const tasks = await callTasksApi(api, todoistQuery);
+	
+	if (tasks.length === 0){
+		new Notice(`Todoist text: You have no tasks matching filter "${todoistQuery}"`);
+	}
+	
+	let returnString = "";
+	if (showSubtasks) {
+		// work through all the parent tasks
+		let parentTasks = tasks.filter(task => task.parentId == null);
+		parentTasks.forEach(task => {
+			returnString = returnString.concat(getFormattedTaskDetail(task, 0, false));
+			returnString = returnString.concat(getSubTasks(tasks, task.id, 1));
+		})
+
+		// display subtasks that have a parent that wasn't returned in the query
+		let orphans:Task[] = [];
+		let subtasks = tasks.filter(task => task.parentId != null);
+		subtasks.forEach(st => {
```

Comment:
a bit nicer (more functional/idiomatic) to write something like:

```ts
const orphans = subtasks.filter(st => !parentTasks.contains(st));
```

Reply:
Thanks!  I figured that there was probably a better way to do that.